### PR TITLE
[SPARK-32934][SQL][FOLLOW-UP] Refine class naming and code comments

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

1. Rename `OffsetWindowSpec` to `OffsetWindowFunction`, as it's the base class for all offset based window functions.
2. Refine and add more comments.
3. Remove `isRelative` as it's useless.

### Why are the changes needed?

code refinement

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

existing tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @beliefer

File: sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExec.scala [hunk 1]
```diff
@@ -57,12 +57,8 @@ import org.apache.spark.sql.types.{CalendarIntervalType, DateType, IntegerType,
...
  *     4. 1 PRECEDING AND 1 FOLLOWING
  *     5. 1 FOLLOWING AND 2 FOLLOWING
- * - Offset frame: The frame consist of one row, which is an offset number of rows. There are three
- * implement of offset frame.
- *     1. [[FrameLessOffsetWindowFunction]] returns the value of the input column offset by a number
- *        of rows according to the current row.
- *     2. [[UnboundedOffsetWindowFunctionFrame]] and [[UnboundedPrecedingOffsetWindowFunctionFrame]]
- *       returns the value of the input column offset by a number of rows within the frame.
+ * - Offset frame: The frame consist of one row, which is an offset number of rows away from the
+ *   current row. Only [[OffsetWindowFunction]]s can be processed in an offset frame.
```
Comment (beliefer) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExec.scala:61]: Do we need to introduce `FrameLessOffsetWindowFunctionFrame`, `UnboundedOffsetWindowFunctionFrame` and `UnboundedPrecedingOffsetWindowFunctionFrame`？

File: sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExecBase.scala [hunk 2]
```diff
@@ -208,7 +209,7 @@ trait WindowExecBase extends UnaryExecNode {
                 target,
                 ordinal,
                 // OFFSET frame functions are guaranteed be OffsetWindowSpec.
```
Comment (beliefer) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExecBase.scala:211]: OffsetWindowSpec -> OffsetWindowFunction

File: sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExecBase.scala [hunk 3]
```diff
@@ -185,7 +186,7 @@ trait WindowExecBase extends UnaryExecNode {
...
```
Comment (beliefer) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExecBase.scala:188]: OffsetWindowSpec -> OffsetWindowFunction

File: sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExecBase.scala [hunk 4]
```diff
@@ -196,7 +197,7 @@ trait WindowExecBase extends UnaryExecNode {
...
```
Comment (beliefer) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExecBase.scala:199]: ditto

**Review by beliefer — APPROVED** [review]

**Comment by beliefer** [comment]
LGTM.

**Comment by cloud-fan** [comment]
merging to master!

**Pull request closed without merging**